src/CKEditor.tsx

```diff
@@ -0,0 +1,209 @@
+/**
+ * @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {
+	eventNameToHandlerName,
+	defaultEvents,
+	stripPrefix,
+	handlerNameToEventName
+} from './events';
+import useCKEditor from './useCKEditor';
+import { camelToKebab } from './utils';
+
+import {
+	CKEditorEventDispatcher,
+	CKEditorEventHandlerProp,
+	CKEditorProps,
+	CKEditorType
+} from './types';
+
+const { useEffect, useRef, useState } = React;
+
+/**
+ * `CKEditor` component is a convenient wrapper around low-level hooks.
+ * It's useful for simpler use cases. For advanced usage see `useCKEditor` hook.
+ */
+function CKEditor<EventHandlerProp>( {
+	config,
+	debug,
+	editorUrl,
+	initData,
+	name,
+	readOnly,
+	style,
+	type,
+
+	/**
+	 * `handlers` object must contain event handlers props only!
+	 */
+	...handlers
+}: CKEditorProps<EventHandlerProp> ): JSX.Element {
+	/**
+	 * Uses `useState` instead of `useRef` to force re-render.
+	 */
+	const [ element, setElement ] = useState<HTMLDivElement | null>( null );
+
+	/**
+	 * Ensures referential equality of event handlers.
+	 */
+	const refs = useRef( handlers );
+
+	const dispatchEvent: CKEditorEventDispatcher = ( { type, payload } ) => {
+		const handlerName = eventNameToHandlerName(
+			stripPrefix( type )
+		) as keyof CKEditorEventHandlerProp;
+		const handler = refs.current[ handlerName ];
+
+		if ( handler ) {
+			handler( payload );
+		}
+	};
+
+	/**
+	 * `readOnly` prop takes precedence over `config.readOnly`.
+	 */
+	if ( config && typeof readOnly === 'boolean' ) {
+		config.readOnly = readOnly;
+	}
+
+	const { editor, status } = useCKEditor( {
+		config,
+		dispatchEvent,
+		debug,
+		editorUrl,
+		element,
+
+		/**
+		 * Subscribe only to those events for which handler was supplied.
+		 */
+		subscribeTo: Object.keys( handlers )
+			.filter( key => key.indexOf( 'on' ) === 0 )
+			.map( handlerNameToEventName ),
+		type
+	} );
+
+	/**
+	 * Sets and updates styles.
+	 */
+	useEffect( () => {
+		const canSetStyles =
+			type !== 'inline' &&
+			editor &&
+			( status === 'loaded' || status === 'ready' );
+
+		if ( style && canSetStyles ) {
+			editor.container.setStyles( style );
+		}
+
+		return () => {
+			if ( style && canSetStyles ) {
+				Object.keys( style )
+					.map( camelToKebab )
+					.forEach( styleName => {
+						editor.container.removeStyle( styleName );
+					} );
+			}
+		};
+	}, [ editor, status, style, type ] );
+
+	/**
+	 * Toggles read-only mode on runtime.
+	 */
+	useEffect( () => {
+		if ( editor && status === 'ready' && typeof readOnly === 'boolean' ) {
+			editor.setReadOnly( readOnly );
+		}
+	}, [ editor, status, readOnly ] );
+
+	return (
+		<div id={name ?? undefined} ref={setElement} style={style ?? undefined}>
+			{initData ? (
+				<div style={{ display: 'none' }}>{initData}</div>
```

Comment:
Doing so you in fact insert invisible `div` into editor's content, which can have weird consequences if connected with certain editor's config, e.g.

```javascript
<CKEditor debug={true} initData="Hello world!" config={{ extraAllowedContent: 'div{*}' }} />
```

will result in "empty" editor (the one containing hidden `div`).

In fact you should just insert raw `initData` into the editor's element.

Reply:
There is another issue with hidden `div`: it changes the way in which SSR works (it does not longer render the visible text).

Reply:
Yeah, that was a hacky way to hide initial content before editor is ready. I need to find a better way. Do you have any hints?

Reply:
I'm not sure if it's even an issue, TBH. CKE4 works that way from the beginning: the content is just visible before editor's initialisation.

Reply:
I've given it one more try. I've added `getStyle` util that returns `style` for the root element based on editor `type` and `status`. It seems to be working fine.